### PR TITLE
Cross seed

### PR DESCRIPTION
**Pull request by RaneyDazed**
# Description

Cross-seed is an app designed to help you download torrents that you can cross seed based on your existing torrents. It is designed to match conservatively to minimize manual intervention.
Cross-seed can inject the torrents it finds directly into your torrent client. Currently the supported clients are
rTorrent
qBittorrent

Link to Project home page: [mmgoodnow/cross-seed](https://github.com/mmgoodnow/cross-seed)
Link to docs : [Cross-seed wiki](https://github.com/mmgoodnow/cross-seed/wiki)

It would be greatly appreciated if you create a sandbox documentation page yourself and do a PR into the [docs repo](https://github.com/saltyorg/docs).  You, as the person creating the role, have presumably used the thing and are presumably familiar with any setup steps required.  Anyone else here would need to research that.

I will do that tomorrow hopefully if I have time. 

# How Has This Been Tested?

I tested it in a VM and the container fires up and runs, fills logs, has correct permissions, creates config.js.

### CODE REVIEW

**Comment by RaneyDazed** [comment]
I guess let me know if i need to change anything..?

**Comment by maximuskowalski** [comment]
Reading through the config it looks like this might need access to torrent directories?

**Comment by RaneyDazed** [comment]
aw sorry max, i did some funky stuff with nano and coder and obviously messed up the formatting. i was between computers at the time. (referencing you doing basic formatting)

**Comment by RaneyDazed** [comment]
should i just add a docker volume for `/mnt/local/downloads/torrents:/torrents`? not sure how else i would specify that.

**Comment by keldian** [comment]
The app expects .torrent files in `/torrents`, and you just told it to find them in `/mnt/local/downloads/torrents`. Could be just me, but I've never seen a .torrent file in there.

I have a feeling you're biting off more than you can chew with this role and before fully understanding how the app works. You're asking if you should do this or that, when instead you should be testing what happens when you do this or that on your end, before even submitting the PR.

The container needs mounts pointing to `/config`, `/torrents` and `/cross-seeds`. None of these are global so they have to be explicitly mapped in the role.

**Comment by RaneyDazed** [comment]
> 

well I assumed that /torrents would be easily mapped to `/torrents/qbittorrent/complete` in config.js. But yes you are probably right. either way, I was asking about that for example, because I thought that leaving it there (at /mnt/local/downloads/torrents) would make it possible for people to specify in the config the exact dir they want to use in the config.

**Comment by keldian** [comment]
Mapping it to `[...]/torrents/qbittorrent/complete` would not amount to anything useful either. There are no .torrent files in there , or at least there shouldn't be. It's where qbittorrent saves its downloaded data, not its .torrent files.

**Comment by RaneyDazed** [comment]
> Mapping it to `[...]/torrents/qbittorrent/complete` would not amount to anything useful either. There are no .torrent files in there , or at least there shouldn't be. It's where qbittorrent saves its downloaded data, not its .torrent files.

@keldian oh then what I followed in the docs for qbittorrent are an edge case I guess. I copy my .torrent files there. I could map it to /opt/qbittorrent? or just /opt. to account for ruTorrent/deluge etc. users.

**Comment by keldian** [comment]
You're making a role. It's up to you to try those things and see if they work. 🤷‍♂

**Comment by owine** [comment]
You could add a list to the Sandbox `settings.yml` file which takes torrent client instances to mount into the container.

```
  clients: ["rutorrent", "qbittorrent", "qbittorrent2"]
```

Add volume mounts as `/opt/{{ clients }}:/{{ clients }}:ro`

**Comment by RaneyDazed** [comment]
> You could add a list to the Sandbox `settings.yml` file which takes torrent client instances to mount into the container.
> 
> ```
>   clients: ["rutorrent", "qbittorrent", "qbittorrent2"]
> ```
> 
> Add volume mounts as `/opt/{{ clients }}:/{{ clients }}:ro`

oh that's awesome. I'll toss that in there in a sec. i had it as "/opt:/opt"

but I'm sure nobody likes having their whole appdata folder open.

**Comment by RaneyDazed** [comment]
Ok. Works great. It should be good now. It has an xseeds dir mapped, torrents are accessed through `/opt/APPNAME/etc...` and config is mapped. I'll go check through the docs again though I may need to adjust it.

File: roles/cross_seed/defaults/main.yml [hunk 1]
```diff
@@ -64,7 +65,8 @@ cross_seed_docker_commands: "{{ cross_seed_docker_commands_default
 # Volumes
 cross_seed_docker_volumes_default:
   - "{{ cross_seed_paths_location }}:/config"
-  - "/mnt/local/downloads/torrents:/torrents"
+  - "/opt:/opt"
```
Comment (owine) [inline roles/cross_seed/defaults/main.yml:68]: Should be `:ro`

Comment (RaneyDazed) [reply roles/cross_seed/defaults/main.yml:68]: Yes, forgot to add that, just a second here

**Comment by maximuskowalski** [comment]
Have you actually used it successfully as well? Normally we can be pretty lax with sandbox but this one has the potential to cause trouble for people who might not want to be booted from from a tracker. I am not sure we need to worry too much but perhaps we might mention this in the docs when we get there. Will see what the workflow says.

**Comment by RaneyDazed** [comment]
I'm currently using it as is with the config filled out. For that reason I'd think you would want to at least touch the config. I didn't explicitly mention getting booted from a tracker, but I suggested a time to wait between checks for cross-seeds.

**Comment by RaneyDazed** [comment]
is it ok for me to comment in the "files changed" directly on the file? `add a comment`?

**Pull request closed without merging**